### PR TITLE
Support multi-arch binaries

### PR DESCRIPTION
**Pull request by yselkowitz**
## Description

Make all binary paths and names contain both OS and ARCH, as a prerequisite of multi-architecture enablement.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Relying on CI to verify.

### CODE REVIEW

**Comment by yselkowitz** [comment]
could you `/ok-to-test` please
cc @robbycochran

File: Makefile [hunk 1]
```diff
@@ -572,7 +576,7 @@ docker-build-data-image: docs-image
 
 .PHONY: docker-build-roxctl-image
 docker-build-roxctl-image:
-	cp -f bin/linux/roxctl image/roxctl/roxctl-linux
+	cp -f bin/linux_$(GOARCH)/roxctl image/roxctl/roxctl-linux
```
Comment (misberner) [inline Makefile:579]: This is referenced as `roxctl-linux-amd64` in `image/roxctl/Dockerfile`, l.4. Curious why this doesn't fail the build - don't we invoke this target?

File: ui/apps/platform/src/Containers/MainPage/Header/CLIDownloadMenu.tsx [hunk 2]
```diff
@@ -37,25 +37,39 @@ function CLIDownloadMenu({ addToast, removeToast }: CLIDownloadMenuProps): React
 
     const appLauncherItems = [
         <ApplicationLauncherItem
-            key="app-launcher-item-cli-mac"
+            key="app-launcher-item-cli-mac-amd64"
```
Comment (misberner) [inline ui/apps/platform/src/Containers/MainPage/Header/CLIDownloadMenu.tsx:40]: It's a bit ugly that all CLI variants are stored in two places. @vjwilson WDYT? @yselkowitz can I push to your PR to clean this up?

File: ui/apps/platform/src/services/CLIService.js [hunk 3]
```diff
@@ -8,9 +8,9 @@ import { saveFile } from 'services/DownloadService';
 export default function downloadCLI(type) {
     let name = 'roxctl';
     let suffix = type;
-    if (type === 'windows') {
+    if (type === 'windows-amd64') {
```
Comment (misberner) [inline ui/apps/platform/src/services/CLIService.js:11]: not your code, but given that who knows what archs we support in the future, I would prefer the more general
```
if (type.startsWith('windows-')) {
  name += 'exe';
  suffix += 'exe';
}
```
up to you

File: tests/upgrade/run.sh [hunk 4]
```diff
@@ -401,8 +401,8 @@ deploy_earlier_central() {
     info "Deploying: $EARLIER_TAG..."
 
     mkdir -p "bin/$TEST_HOST_OS"
-    gsutil cp "gs://stackrox-ci/roxctl-$EARLIER_TAG" "bin/$TEST_HOST_OS/roxctl"
-    chmod +x "bin/$TEST_HOST_OS/roxctl"
+    gsutil cp "gs://stackrox-ci/roxctl-$EARLIER_TAG" "bin/${TEST_HOST_OS}_amd64/roxctl"
+    chmod +x "bin/${TEST_HOST_OS}_amd64/roxctl"
     PATH="bin/$TEST_HOST_OS:$PATH" command -v roxctl
```
Comment (misberner) [inline tests/upgrade/run.sh:406]: You need to adjust these lines to `${TEST_HOST_OS}_amd64` as well. The problem is that we copy the binary to `linux_amd64`, but we don't add it to the path, so the deploy scripts don't find the roxctl binary and thus assume we need to run docker to invoke roxctl, which doesn't work in CI. This is the cause of the upgrade tests failure.

**Review by misberner — CHANGES_REQUESTED** [review]
This looks pretty good overall! I think the upgrade tests failure is real and I have commented below on what I think needs to be done.

What is still missing is the publishing logic (i.e., uploading the CLI to stackrox.io and openshift mirror). You find the relevant scripts in `scripts/ci/roxctl-publish/` (@gavin-stackrox can you briefly confirm that this is still the relevant location, even in the OSCI world?), and I think that `prepare.sh` is the one of interest, as `publish.sh` just works on the directory structure left behind by `prepare.sh`.

I would suggest to change things in such a way that for now, amd64 binaries without the arch suffix are still published in addition to the arch-specific names. The reason is that download links are referenced in the docs, and we cannot upgrade the docs atomically with releasing the new version. Therefore, make sure we have a transition period in which we can fix the docs (note to self: check if there's a JIRA for doc changes).

File: ui/apps/platform/src/Containers/MainPage/Header/CLIDownloadMenu.tsx (hunk 2 above)
Comment (vjwilson) [reply ui/apps/platform/src/Containers/MainPage/Header/CLIDownloadMenu.tsx:40]: @misberner 
The `key` properties are an affordance for the React templating system. They just need to be unique IDs. In fact, let's not try to keep them in sync with variant in the click handler--that's gilding a lily. Just make them the type of system

For example,
* mac
* linux-amd
* linux-ppc

Comment (misberner) [reply ui/apps/platform/src/Containers/MainPage/Header/CLIDownloadMenu.tsx:40]: Oh, I was referring to it being referenced in the other file as well.

**Review by misberner — COMMENTED** [review]
@yselkowitz FYI, we only allow merging PRs via "Rebase & squash" to this repo, so you can accumulate an arbitrary number of commits on your branch with no effect on the `master` branch post merging. I would suggest to force push only when rebasing, as otherwise I lose the ability to do incremental reviews on GitHub.

**Comment by gavin-stackrox** [comment]
@misberner yes, the scripts in `scripts/ci/roxctl-publish` are still used to publish roxctl on release.

**Comment by misberner** [comment]
@yselkowitz please rebase to fix CI failures

**Comment by misberner** [comment]
@yselkowitz FYI I've for now decided against pushing the feature flags changed against your branch, I think we can easily take care of it at a later point. Let's focus on getting this merged first.

**Comment by misberner** [comment]
Also, the changes look good to me! Let's see what CI says..

**Review by misberner — APPROVED** [review]

**Comment by misberner** [comment]
Awesome. All tests passed. 🚢 it

**Comment by misberner** [comment]
@yselkowitz tbh, I don't know how our repo is set up. Can you merge this yourself, or do you need me to do it?

**Pull request merged**